value/sys.go

```diff
@@ -39,53 +43,134 @@ To convert a time vector to a seconds value:
+		Errorf("sys %q not defined", verb)
+	}
+
+	Errorf("sys requires string argument")
+	panic("unreachable")
+}
+
+var sys1 = map[string]func(c Context) Value{
+	"help": func(c Context) Value {
+		fmt.Fprint(c.Config().Output(), sysHelp)
 		return empty
-	case "base":
-		return NewIntVector(conf.Base())
-	case "cpu":
-		real, user, sys := conf.CPUTime()
+	},
+	"base": func(c Context) Value {
+		return NewIntVector(c.Config().Base())
+	},
+	"cpu": func(c Context) Value {
+		real, user, sys := c.Config().CPUTime()
 		vec := make([]Value, 3)
 		vec[0] = BigFloat{big.NewFloat(real.Seconds())}
 		vec[1] = BigFloat{big.NewFloat(user.Seconds())}
 		vec[2] = BigFloat{big.NewFloat(sys.Seconds())}
 		return NewVector(vec)
-	case "date":
-		return newCharVector(time.Now().In(conf.Location()).Format(time.UnixDate))
-	case "format":
-		return newCharVector(fmt.Sprintf("%q", conf.Format()))
-	case "ibase":
-		return Int(conf.InputBase())
-	case "maxbits":
-		return Int(conf.MaxBits())
-	case "maxdigits":
-		return Int(conf.MaxDigits())
-	case "maxstack":
-		return Int(conf.MaxStack())
-	case "obase":
-		return Int(conf.OutputBase())
-	case "origin":
-		return Int(conf.Origin())
-	case "prompt":
-		return newCharVector(fmt.Sprintf("%q", conf.Prompt()))
-	case "sec", "now":
+	},
+	"date": func(c Context) Value {
+		return newCharVector(time.Now().In(c.Config().Location()).Format(time.UnixDate))
+	},
+	"format": func(c Context) Value {
+		return newCharVector(fmt.Sprintf("%q", c.Config().Format()))
+	},
+	"ibase": func(c Context) Value {
+		return Int(c.Config().InputBase())
+	},
+	"maxbits": func(c Context) Value {
+		return Int(c.Config().MaxBits())
+	},
+	"maxdigits": func(c Context) Value {
+		return Int(c.Config().MaxDigits())
+	},
+	"maxstack": func(c Context) Value {
+		return Int(c.Config().MaxStack())
+	},
+	"now": func(c Context) Value {
 		return BigFloat{big.NewFloat(float64(time.Now().UnixNano()) / 1e9)}
-	case "time":
-		return timeVec(time.Now().In(conf.Location()))
-	default:
-		Errorf("sys %q not defined", arg)
+	},
+	"obase": func(c Context) Value {
+		return Int(c.Config().OutputBase())
+	},
+	"origin": func(c Context) Value {
+		return Int(c.Config().Origin())
+	},
+	"prompt": func(c Context) Value {
+		return newCharVector(fmt.Sprintf("%q", c.Config().Prompt()))
+	},
+	"sec": func(c Context) Value {
+		return BigFloat{big.NewFloat(float64(time.Now().UnixNano()) / 1e9)}
+	},
+	"time": func(c Context) Value {
+		return timeVec(time.Now().In(c.Config().Location()))
+	},
+}
+
+var sys2 = map[string]func(Context, []Value) Value{
+	"read": sysRead,
+}
+
+func sysRead(c Context, args []Value) Value {
+	usage := func() {
+		Errorf(`usage: sys "read" "filename"`)
+	}
+
+	if len(args) != 1 {
+		usage()
+	}
+	v, ok := args[0].(Vector)
+	if !ok || !allChars(v) {
+		usage()
+	}
+	file := vecText(v)
+
+	f, err := os.Open(file)
+	if err != nil {
+		Errorf("%v", err)
+		return empty
+	}
+	defer f.Close()
+
+	var out Vector
+	r := bufio.NewReader(f)
+	for {
```

Comment:
for minor consistency, i'd prefer you to use Scanner over Reader.

Reply:
Done.